feature/experimental/gribi/ate_tests/dut_daemon_failure/dut_daemon_failure_test.go

```diff
@@ -0,0 +1,393 @@
+	for _, entry := range entries {
+		v := entry.Entry.(*grps.AFTEntry_Ipv4)
+		if prefix := v.Ipv4.GetPrefix(); prefix != "" {
+			if prefix == ateDstNetCIDR {
+				found = true
+				t.Logf("Found route to %s in gRIBI Get response", ateDstNetCIDR)
+				break
+			}
+		}
+	}
+	if !found {
+		t.Errorf("Route to %s NOT found in gRIBI Get response", ateDstNetCIDR)
+	}
+}
+
+// gNOIKillProcess kills a daemon on the DUT, given its name and pid.
+func gNOIKillProcess(ctx context.Context, t *testing.T, args *testArgs, pName string, pID uint32) {
+	gnoiClient := args.dut.RawAPIs().GNOI().Default(t)
+	killRequest := &gnps.KillProcessRequest{Name: pName, Pid: pID, Signal: gnps.KillProcessRequest_SIGNAL_TERM, Restart: true}
+	killResponse, err := gnoiClient.System().KillProcess(context.Background(), killRequest)
+	t.Logf("Got kill process response: %v\n\n", killResponse)
+	if err != nil {
+		t.Fatalf("Failed to execute gNOI Kill Process, error received: %v", err)
+	}
+}
+
+func TestDUTDaemonFailure(t *testing.T) {
+
+	start := time.Now()
+	dut := ondatra.DUT(t, "dut")
+	ctx := context.Background()
+	var flow *ondatra.Flow
+
+	// Configure the DUT.
+	t.Logf("Configure DUT")
+	configureDUT(t, dut)
+
+	// Configure the ATE.
+	t.Logf("Configure ATE")
+	ate := ondatra.ATE(t, "ate")
+	top := configureATE(t, ate)
+	top.Push(t).StartProtocols(t)
+
+	t.Logf("Time check: %s", time.Since(start))
+
+	args := &testArgs{
+		ctx: ctx,
+		dut: dut,
+		ate: ate,
+		top: top,
+	}
+
+	t.Run("SetupGribiConnection", func(t *testing.T) {
+
+		// Set parameters for gRIBI client clientA.
+		// Set Persistence to true.
+		clientA := &gribi.Client{
+			DUT:                  dut,
+			FibACK:               false,
+			Persistence:          true,
+			InitialElectionIDLow: 10,
+		}
+
+		t.Log("Establish gRIBI client connection")
+		if err := clientA.Start(t); err != nil {
+			t.Fatalf("gRIBI Connection for clientA could not be established")
+		}
+
+		t.Run("AddRoute", func(t *testing.T) {
+			t.Logf("Add gRIBI route to %s and verify through Telemetry and Traffic", ateDstNetCIDR)
+			addRoute(ctx, t, args, clientA)
+
+			t.Run("VerifyAFT", func(t *testing.T) {
+				verifyAFT(ctx, t, args)
+			})
+
+			t.Run("VerifyTraffic", func(t *testing.T) {
+				verifyTraffic(ctx, t, args)
+			})
+
+		})
+
+		t.Logf("Time check: %s", time.Since(start))
+
+	})
+
+	t.Run("RestartTraffic", func(t *testing.T) {
+
+		srcEndPoint := top.Interfaces()[atePort1.Name]
+		dstEndPoint := top.Interfaces()[atePort2.Name]
+
+		flow = startTraffic(t, args.ate, args.top, srcEndPoint, dstEndPoint)
+
+		t.Logf("Wait for 15 seconds")
+		time.Sleep(15 * time.Second)
+
+	})
+
+	t.Run("KillGribiDaemon", func(t *testing.T) {
+
+		// Find the PID of gRIBI Daemon.
+		var pName string = "unknown"
+		var pId uint64
+		t.Run("FindGribiDaemonPid", func(t *testing.T) {
+			switch dut.Vendor() {
+			case ondatra.JUNIPER:
+				pName = "rpd"
+			case ondatra.CISCO:
+				pName = "emsd"
+			case ondatra.ARISTA:
+				pName = "Gribi"
+			}
```

Comment:
We have pretty good vendor coverage here, but we probably need to think of a way that makes it obvious how to extend the test for future vendor coverage rather than relying on someone to recognize the meaning of the "unknown" placeholder string.

I suggest moving the switch statement to a map, and do a check at the beginning of the test that the vendor is supported. The idea is to do the check that we can do quickly before doing something more time consuming.

```
// gribiDaemon is the vendor-specific name for the gRIBI process name.
var gribiDaemon = map[ondatra.Vendor]string{
  ondatra.ARISTA: "Gribi",
  ondatra.CISCO: "emsd",
  ondatra.JUNIPER: "rpd",
}

func TestDUTDaemonFailure(t *testing.T) {
  dut := ondatra.DUT(t, "dut")
  if _, ok := gribiDaemons[dut.Vendor()]; !ok {
    t.Fatalf("Please add support for vendor %v to gribiDaemons", dut.Vendor())
  }
  ...
}
```